src/java_bytecode/ci_lazy_methods_needed.h

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************\
+
+Module: Context-insensitive lazy methods container
+
+Author: Chris Smowton, chris.smowton@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Context-insensitive lazy methods container
+
+#ifndef CPROVER_JAVA_BYTECODE_CI_LAZY_METHODS_H
+#define CPROVER_JAVA_BYTECODE_CI_LAZY_METHODS_H
+
+#include <vector>
+#include <set>
+#include <util/symbol_table.h>
+
+class ci_lazy_methods_neededt
+{
+public:
+  ci_lazy_methods_neededt(
+    std::vector<irep_idt> &_needed_methods,
+    std::set<irep_idt> &_needed_classes,
+    symbol_tablet &_symbol_table):
+  needed_methods(_needed_methods),
+  needed_classes(_needed_classes),
+  symbol_table(_symbol_table)
+  {}
+
+  void add_needed_method(const irep_idt &);
```

Comment:
You mean the fact `add_needed_class` returns a boolean? I agree - I will see if it is easy to add a sensible boolean to the `add_needed_method` (it is also a bit odd that one is stored in a set, and one is stored in a vector...)